src/components/Charts/AnimatedInLineChart.tsx

```diff
@@ -0,0 +1,90 @@
+import { Group } from '@visx/group'
+import { LinePath } from '@visx/shape'
+import { easeCubicInOut } from 'd3'
+import React from 'react'
+import { useEffect, useRef, useState } from 'react'
+import { animated, useSpring } from 'react-spring/web'
+import { useTheme } from 'styled-components/macro'
+
+import { LineChartProps } from './LineChart'
+
+const config = {
+  duration: 800,
+  easing: easeCubicInOut,
+}
+
+// code reference: https://airbnb.io/visx/lineradial
+
+function LineChart<T>({
+  data,
+  getX,
+  getY,
+  marginTop,
+  curve,
+  color,
+  strokeWidth,
+  width,
+  height,
+  children,
+}: LineChartProps<T>) {
+  const lineRef = useRef<SVGPathElement>(null)
+  const [lineLength, setLineLength] = useState<number>(0)
+  const [shouldAnimate, setShouldAnimate] = useState<boolean>(false)
+  const [hasAnimatedIn, setHasAnimatedIn] = useState<boolean>(false)
+
+  const spring = useSpring({
+    frame: shouldAnimate ? 0 : 1,
+    config,
+    onRest: () => {
+      setShouldAnimate(false)
+      setHasAnimatedIn(true)
+    },
+  })
+
+  const effectDependency = lineRef.current
+  useEffect(() => {
+    if (lineRef.current) {
+      setLineLength(lineRef.current.getTotalLength())
+      setShouldAnimate(true)
+    }
+  }, [effectDependency])
+  const theme = useTheme()
+  const lineColor = color ?? theme.accentAction
+
+  return (
+    <svg width={width} height={height}>
+      <Group top={marginTop}>
+        <LinePath curve={curve} x={getX} y={getY}>
+          {({ path }) => {
+            const d = path(data) || ''
+            return (
+              <>
+                <animated.path
```

Comment:
Why can't you use a single path and change the stroke's opacity, dashoffset, and dasharray with animation? I think that would let you reduce state and simplify.

Reply:
you need to measure the length of the line in order to animate along it, so basically you need to do a render first. the code is from here https://airbnb.io/visx/lineradial which references this https://www.visualcinnamon.com/2016/01/animating-dashed-line-d3/

Reply:
is this initial render causing the flash on hard refresh, as noted by @infredible ?

Reply:
Can you explain this in a comment, where you currently have the reference?